extensions/src/main/java/org/pentaho/platform/plugin/action/ActionResourceUtil.java

```diff
@@ -0,0 +1,72 @@
+/*!
+ * This program is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License, version 2.1 as published by the Free Software
+ * Foundation.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along with this
+ * program; if not, you can obtain a copy at http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ * or from the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ * Copyright (c) 2017 Pentaho Corporation..  All rights reserved.
+ */
+package org.pentaho.platform.plugin.action;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.pentaho.platform.api.action.ActionInvocationException;
+import org.pentaho.platform.api.action.IAction;
+import org.pentaho.platform.api.action.IActionInvokeStatus;
+import org.pentaho.platform.api.action.IActionInvoker;
+import org.pentaho.platform.plugin.action.messages.Messages;
+import org.pentaho.platform.util.ActionUtil;
+import org.pentaho.platform.util.StringUtil;
+import org.slf4j.MDC;
+
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Map;
+import java.util.Optional;
+
+public class ActionResourceUtil {
+  protected static final Log logger = LogFactory.getLog( ActionResourceUtil.class );
+
+  private static IAction createActionBean( final String actionClass, final String actionId ) throws Exception {
+    return ActionUtil.createActionBean( actionClass, actionId );
+  }
+
+  private static Map<String, Serializable> deserialize( IAction action, final String actionParams ) throws IOException, ActionInvocationException {
+    return ActionParams.deserialize( action, ActionParams.fromJson( actionParams ) );
+  }
+
+  public static IActionInvokeStatus runWorkItemRequest( String actionClass, String actionId, String actionParams, String user, IActionInvoker actionInvoker, Map< String, String > mdcContextMap ) {
```

Comment:
I named it work item request to avoid confusion because the method contract is different than the API invokeAction